Add neutron validation runs (#599)
legend-exp/remage: docs/validation/neutrons.md

# Neutron transport and capture spectrum

Here we investigate the neutron capture physics. For this, there are four

> different setups.

Reviewer: maybe you could add links to the relevant remage docs

https://remage.readthedocs.io/en/stable/manual/physicslist.html#hadronic-physics-options

https://remage.readthedocs.io/en/stable/api/classRMGNeutronCaptureProcess.html#_CPPv424RMGNeutronCaptureProcess

Reviewer 2: unfortunately we do not have a good way to link back, so this is a bit fragile. We cannot use the typical references as in the rest of the documentation... It might also be a bit bad that this would always link to the stable docs, and not to the version the validation suite is built for?

> - Simple world filled with `"G4_STAINLESS-STEEL"`. The only active process for

Reviewer: can we move these descriptions to the relative sections? i think it would make things clearer

  neutrons is `nCapture` and neutrons are sampled at 10 keV. To compare the

>   de-excitation gammacascades against other models.

Reviewer: ```suggestion
  de-excitation gamma cascades against other models.
```

- Simple world, filled with `"G4_Gd"`. The only active process for neutrons is
  `nCapture` and neutrons are sampled at 0.1 eV. To compare the capture
  distribution and check that the _remage_ custom `RMGnCapture` process produces
  the same result as the native Geant4.
- Simple world, filled with `"G4_Gd"`. The only active process for neutrons is
  `nCapture` and neutrons with a log-uniform distribution from 0.01 eV to 1 MeV.

>   To re-create the relative cross-section of nCapture on gadolinium and compare

Reviewer: ```suggestion
  To re-create the relative cross-section of `nCapture` on gadolinium and compare
```

  against literature.
- Three simple worlds, filled with liquid Argon that consists only of one of the
  stable Argon isotopes. Every process is active. Neutrons are sampled with a
  log-uniform distribution from 0.01 eV to 10 MeV. To re-create the total
  cross-section for neutrons in argon.
- Complex world, containing Argon, Gd-loaded moderator, Steel and Water.
  Neutrons are sampled at 5 keV but every process is active. To validate in a
  more realistic scenario.

> ## Simple steel world

Reviewer: Maybe better sth like "captures on steel"?

> Here, the gamma spectrum after the nCapture is shown. The goal is to validate or

Reviewer: ```suggestion
Here, the gamma spectrum after the `nCapture` is shown. The goal is to validate or
```

compare this against other models, which are missing for now. As previously
mentioned: Simple world filled with `"G4_STAINLESS-STEEL"`. The only active
process for neutrons is `nCapture` and neutrons are sampled at 10 keV.

> ```{figure} ./_img/neutrons/neutron_spectrum_steel_shielding.output.png

Reviewer: use widths that are fraction of the page everywhere

Reviewer: i don't understand the offsets of the histograms in this plot

Author: The offsets are inspired by the plot grabmayr sent to me and i think it looks good to just have all isotopes in one singular plot. As the total height of the histogram does not matter, it is only about the intensities. Between every isotope there is an offset of 1e5. That number was just arbitrarily chosen by eye, regarding what looks best.

Bigger number obviously means more empty space between lines, which in turn means less height for each isotope (if you keep the figure dimensions).

Reviewer: it's ok but it's not explained anywhere atm?

Author: ah ok

:width: 800px
```

Additionally a pie-chart showing the distribution of the nCaptures can be shown.

> This can be used to validate the nCapture cross-sections of steel at 10 keV

Reviewer: ```suggestion
This can be used to validate the `nCapture` cross-sections of steel at 10 keV
```
check formatting everywhere

> ```{figure} ./_img/neutrons/neutron_pie_chart_steel.output.png

Reviewer: this figure is rendered quite big, can you set it to 50% maybe?

:width: 800px
```

> ## Geant4 nCapture vs RMGnCapture

Reviewer: ```suggestion
## Geant4 `nCapture` vs `RMGnCapture`
```

These pie-charts show the capture distribution on the gadolinium isotopes. They
should be the same in both cases.

```{subfigure} AB
:subcaptions: above

:::{image} ./_img/neutrons/neutron_pie_chart_normal.output.png
:width: 90%
:alt: Distribution using the Geant4 nCapture.
:::

:::{image} ./_img/neutrons/neutron_pie_chart_rmgncapture.output.png
:width: 90%

> :alt: Distribution using the custom RMGnCapture.

Reviewer: ```suggestion
:alt: Distribution using the custom `RMGnCapture`.
```

:::

> If both plots do not show the same thing (within std) there is an issue.

Reviewer: what std? what is std?

```

We can also take a look at the de-excitation gamma spectrum:

```{figure} ./_img/neutrons/neutron_spectrum_gd_shielding.output.png
:width: 800px
```

## Relative cross-section of gadolinium

> First we take a look at the latest G4NDL data that is stored for Geant4, to see

Reviewer: is this taken from the actual geant4 installation the validation suite is run with?

Author: yes, i clarified it in the text

if it exists and is reasonable.

```{figure} ./_img/neutrons/neutron_gd_G4NDL.output.png
:width: 800px
```

Next, the relative nCapture cross-section for gadolinium is reconstructed and
compared to the nCapture cross-section according to JEFF-3.3. Each isotope was
weighted by the natural abundance in gadolinium and the relative fraction of the
total nCapture cross-section was derived. Additionally, the data was re-binned
to fit to the binning of the simulation. The relative cross-section from the
simulation was re-constructed by counting the number of captures on each isotope
at given energy.

```{figure} ./_img/neutrons/neutron_gd_cross_section.output.png
:width: 800px
```

## Argon transport

First, the total cross-section for neutrons with the different argon isotopes is
compared. For this, Geant4 typically uses G4NDL data. In the following plots,
the latest G4NDL data is directly (dynamically) extracted out of the Geant4 data

> folder within the remage container. The solid lines shows the static JEFF-3.3

Reviewer: ```suggestion
the latest G4NDL data is directly (dynamically) extracted out of the data
folder part of the Geant4 installation. The solid lines shows the static JEFF-3.3
```

data (on which at least G4NDL4.7 is based on), while the corresponding dots
(squares) show the latest G4NDL data. It can be possible, that G4NDL data is
missing for certain isotopes. This is indicated by the label stating "missing:
...". The letters stand for : "C" means the "Capture" data is missing, "E" means
the "elastic scattering" data is missing and "I" means the "inelastic
scattering" data is missing.

```{figure} ./_img/neutrons/neutron_jeff_vs_g4ndl.output.png
:width: 800px
```

> To not only verify the datafiles, but also that they get applied correctly and

Reviewer: ```suggestion
To not only verify the data files, but also that they get applied correctly and
```

the simulation works, the total cross-section for neutrons in liquid Argon is
reconstructed from an idealized simulation. In this idealized simulation an orb
of 2 km radius is used. There are three worlds and three simulations, one for
each argon isotope. The density (and temperature) of the argon is that of liquid

> argon. To calculate the total cross-sections, first the steplength between two

Reviewer: ```suggestion
argon. To calculate the total cross-sections, first the step length between two
```

steps is derived, using the position of each step. The mean of this value is
taken as mean free path and used to derive the total cross-section. The energy
is given by the velocity of the particle at pre-step. Due to the 2 km radius of
the orb, geometric boundaries should not be relevant. Additionally, a distance
to surface check only counts those events, that end up at least 10 m away from
any boundary. Still, the results look like there might be a bias from the
simulation. This could be caused by non-physical simulation processes limiting

> the steplength, but has not been investigated so far.

Reviewer: add a TODO admonition box to say that this should be investigated with future studies

:width: 90%
:alt: argon-36 total cross-section.
:::

:::{image} ./_img/neutrons/neutron_argon38_comparison.output.png
:width: 90%
:alt: argon-38 total cross-section.
:::

:::{image} ./_img/neutrons/neutron_argon40_comparison.output.png
:width: 90%
:alt: argon-40 total cross-section.
:::

The total cross-sections of the three stable argon isotopes.
```

For cases where the G4NDL cross-section is missing, Geant4 might fall back to
theoretical models.

## Complex geometry

In this setup neutrons are simulated in a steel container filled with liquid
argon. The steel container is located within a water volume and inside of the
argon there is also a neutron moderator, which consists of hydrogen and is
loaded with gadolinium. The goal of the geometry is to enable not only neutron
capture on different isotopes, but also have the neutron transport be a crucial
component regarding capture distribution. The main elements on which can be
captured in this geometry are therefore H, Ar, Cr, Fe, Ni, Gd.

First, the pie-chart showing where the captures took place is shown:

```{figure} ./_img/neutrons/neutron_pie_chart_general.output.png

> :width: 800px

Reviewer: also very big figure, reduce to 50% width

this setup. The expected Q-value according to the
[IAEA Nuclear Data Section](https://www-nds.iaea.org/) is shown in each plot for
the respective isotope. This can directly be compared with the orange line,
depicting the summed energy of all secondary particles resulting from the
neutron capture process in Geant4. In blue, the gamma line intensity is shown.

For every isotope there are always two plots shown. The plot on the left just
specifies the basic `G4Shielding` physicslist to the simulation without any
extra modifications. The plot on the right instead additionally forces the
`G4PhotonEvaporation` model to be used by specifying the macro command
`/process/had/particle_hp/use_photo_evaporation true`. The `G4PhotonEvaporation`
model should always correctly respect the Q-value of the capture, but is very
likely to not correctly represent the gamma line intensity. The standard
`G4Shielding` option might fall back to the `G4PhotonEvaporation` depending on
the available data and version. Both plots might show a different number of
captures per isotope due to the random engine getting out of sync. Still numbers
should be within standard deviation of another.

> Only these two models are shown due to run-time constraints. Due to the

Reviewer: is this really the maximum stats we can have on the validation suite runner in a reasonable amount of time? @ManuelHu

Author: This test went from ~250s to ~110s on my local pc after the multi-threading change. The runtime did not significantly change here on the runner after that change (i checked and it actually took longer on mac for example).

This tells me that this validation part is not significantly contributing to the runtime. So i can increase the statistics. But it gets problematic quick i guess.

It should be quicker after the option from Moritz PR to kill all secondaries in the stepping. But after that i do not think there is any way to speed it up more

# Neutron transport and capture spectrum

Here we investigate the neutron physics. _remage_ allows for the specification
of hadronic physics with the macro command
<project:../rmg-commands.md#rmgprocesseshadronicphysics> as explained in
<project:../manual/physicslist.md#hadronicphysicsoptions>. This validation will
focus on the `Shielding` option for hadronic physics. Additionally, _remage_
allows for a modification of the neutron capture process according to
<project:../manual/physicslist.md#customgrabmayrgammacascades>, which will also
be validated. For this, there are five different main parts.

> 

Reviewer 3: I think it would be useful for general audience readers to provide a very brief overview of the neutron physics implementation in G4, i.e., what the idea behind Neutron_HP/Particle_HP is and what G4NDL and evaluated nuclear data libraries are. Also, maybe a quick reminder about the relevant processes for neutrons in matter (elastic/inelastic scattering, radiative capture) and how their cross sections look (e.g., you reference the resonance-area below without explanation). This information is already scattered throughout the document, but I think it would be better to explain it in the introduction section.

Also, maybe a bit of motivation on why we are interested in validating this with a reference to L1000, and why it is interesting to understand the materials you tested.

Reviewer 2: another comment on this section: references to the normal documentation(e.g. with `<project:../` will not work in the validation suite.

Author: Okay, i added a small abstract to the introduction part. But i did not go into the relevant processes for neutrons in matter, as i kinda feel like the validation suite is not the place to explain basic physics... I might be wrong idk

Reviewer 3: It would be nice to have that, or at least pointers to where people can read more about it, i.e., links to the relevant pages in the G4 documentation.

versions.

The neutron capture gamma de-excitation can be validated against literature
values. Hydrogen has a well-known singular gamma line at 2.22 MeV after a
neutron capture, making bugs very easy to spot. Gadolinium on the contrary, has
a very complex gamma line intensity, which can be compared to
[publications](https://link.springer.com/article/10.1140/epjc/s10052-023-11602-y).
The Q-Values of the neutron capture process is well known for all captures in
this setup. The expected Q-value according to the
[IAEA Nuclear Data Section](https://www-nds.iaea.org/) is shown in each plot for
the respective isotope. This can directly be compared with the orange line,
depicting the summed energy of all secondary particles resulting from the
neutron capture process in Geant4. In blue, the gamma line intensity is shown.

For every isotope there are always two plots shown. The plot on the left just
specifies the basic `G4Shielding` physicslist to the simulation without any
extra modifications. The plot on the right instead additionally forces the
`G4PhotonEvaporation` model to be used by specifying the macro command

> `/process/had/particle_hp/use_photo_evaporation true`. The `G4PhotonEvaporation`

Reviewer 3: You might also want to test for `/process/had/particle_hp/skip_missing_isotopes true` and `/process/had/particle_hp/do_not_adjust_final_state true` since they are highly recommended by the [G4ApplicationBook](https://geant4-userdoc.web.cern.ch/UsersGuides/ForApplicationDeveloper/html/TrackingAndPhysics/physicsProcess.html?highlight=adjust#high-precision-neutron-interactions-neutronhp)). When the former is not set (default), G4 will use a isotope similar to the missing one insted which we definetly want to avoid. When the latter is not set (default), G4 will emit additional gammas to preserve energy conservation. For the latter, there is an argument to keep it set to false, though.

Author: What change do you have in mind? I would like to avoid adding even more simulations. These tests already consist out of 9 simulations and every added sim takes statistics from the others (to stay reasonable in time).
For the first one i think we can make an argument to set it to true just by remage default? 
Does it affect the gamma cascades or also the capture cross-section?

Reviewer 3: It should only affect the cross sections:
> /process/had/particle_hp/skip_missing_isotopes true
>
>    This UI commands sets to zero the cross section of the isotopes which are not present in the neutron library. If Geant4 doesn’t find an isotope, then it looks for the natural composition data of that element. Only if the element is not found then the cross section is set to zero. On the contrary, if this variable is not defined, Geant4 looks then for the neutron data of another isotope close in Z and A, which will have completely different nuclear properties and lead to incorrect results (highly recommended).

I think it would be useful to check whether G4 actually sets the cross section to 0 and doesn't nick it from a completely unrelated isotope. Maybe that might be also what we are seeing with Ar36 and Ar38?

For the second command, it is a question of whether we want to conserve individual events' energy (false) or ensure that the statistical results are consistent with G4NDL (true). The false state is probably more what we want.

Regarding the statistics: I thought that with the kill-all-secondary implementation, the statistics would no longer be an issue. You may also be able to get more out of the multiprocessing or multithreading of individual simulations (we have 32 cores on the server after all).

Author: I think that probably is what we are seeing with Ar36 and Ar38, although i think they removed the warnings?
In [this post](https://geant4-forum.web.cern.ch/t/why-does-the-neutronhp-model-in-geant4-v11-2-2-give-warnings-about-missing-files-for-argon-isotopes-when-this-issue-did-not-occur-in-geant4-v11-2-1-and-v11-0-3-and-how-can-it-be-resolved/12786) it shows that.

Well, we will see the impact of the kill-all-secondary command. Multithreading the individual simulation might help currently (as the neutron simulations run 500 sec out of the total 699.01 sec. ). But this also comes with overhead again. And i think if there are going to be just more simulations, then just having each simulation run single threaded but in parallel (the way it is done now) is better.

installation of _remage_. As of `G4NDL4.7.1` these are mostly based on JEFF-3.3.

:::{note}

The installation of new neutron data libraries should work the same way as for
basic Geant4. Copy the libraries in the corresponding folders (which should
either be under `"/opt/geant4/share/Geant4/data"` or under
`$CONDA_PREFIX/share/Geant4/data`, depending on the type of installation) and
point the `G4NEUTRONHPDATA` variable to them.

:::

_remage_ allows for the specification of hadronic physics with the macro command
[`/RMG/Processes/HadronicPhysics`](https://remage.readthedocs.io/en/stable/rmg-commands.html#rmg-processes-hadronicphysics)
as explained in the
[Hadronic Physics Options](https://remage.readthedocs.io/en/stable/manual/physicslist.html#hadronic-physics)
section. This validation will focus on the `Shielding` option for hadronic
physics. Additionally, _remage_ allows for a modification of the neutron capture
process according to
[this section](https://remage.readthedocs.io/en/stable/manual/physicslist.html#custom-grabmayr-gamma-cascades),
which will also be validated.

The materials tested in these validations are chosen based on their relevance

> for particle physics experiments, such as LEGEND-1000.

Reviewer 3: On that note, if we argue that it is relevant to muon-induced isotope production, we should also test the cross sections for O and C in PMMA and (maybe significantly more relevant) for Ge in HPGe. 

In the future, we should also overlay recent experimental data from https://www-nds.iaea.org/exfor/ to compare with both simulation/G4NDL to check consistency.

Reviewer 3: I should clarify that these do not necessarily have to be implemented in this PR, just as a note for the future.

Author: I think C and O is negligible? At least from experience i have barely seen anything happen on C and O (which is biased as that might be due to Geant4 being wrong, lol) (Also PMMA and with that C and O is in the more complex test). But it might be reasonable to also add a test for this. But i am somewhat always very deterred from flooding the validation suite with tests.

So maybe this is also a different question on whether we want the validation suite to focus only on core aspects or if we want it to be as extensive as possible.

Regarding Ge i totally agree. But i think for that case we might want to put more effort into it than just a basic comparison? (As i think you have done a lot of work, scaling it to other data etc.).

Anyways i am going to add a todo to the file, as i do not plan on adding more tests to this PR

isotope is shifted up by a constant offset.

```{figure} ./_img/neutrons/neutron_spectrum_steel_shielding.output.png
:width: 70%
```

:::{todo}

Add spectra of other models for comparison

:::

Additionally, a pie-chart showing the distribution of the neutron captures
results as a by-product and can be shown. This can be used to validate the
`nCapture` cross-sections of steel at 10 keV. The material `G4_STAINLESS-STEEL`
is made out of 74% iron, 18% chromium and 8% nickel. Despite iron consisting to
91.8% out of Fe-56, we would not necessarily expect Fe-56 to have the main
contribution at 10 keV. This is because 10 keV is right in the resonance-area
and the cross section heavily depends on the actual database you are using. The
same applies to the other elements.

```{figure} ./_img/neutrons/neutron_pie_chart_steel.output.png
:width: 50%
```

## Geant4 `nCapture` vs `RMGnCapture`

In this simple world filled with `"G4_Gd"`, the only active process again is
either `nCapture` or `RMGnCapture`. The neutrons are sampled at 0.1 eV. The
pie-charts show the capture distribution on the gadolinium isotopes. They are
expected to show the same thing in both cases, as the `RMGnCapture` process

> should only impact the gamma cascade and not the capture cross-section.

Reviewer 3: Consider giving a quick reminder on why we have a wrapper for the nCapture process to motivate this test.

total `nCapture` cross-section was derived. Additionally, the data was re-binned
to fit to the binning of the simulation. The relative cross-section from the
simulation was re-constructed by counting the number of captures on each isotope
at given energy.

```{figure} ./_img/neutrons/neutron_gd_cross_section.output.png
:width: 70%
```

## Argon transport

Here, we validate neutron transport in argon. First, the total cross-section for
neutrons with the different argon isotopes is compared. For this, Geant4
typically uses G4NDL data. In the following plots, the latest G4NDL data is
directly (dynamically) extracted out of the data folder part of the Geant4
installation. The solid lines shows the static JEFF-3.3 data (on which at least
G4NDL4.7 is based on), while the corresponding dots (squares) show the latest
G4NDL data. It can be possible, that G4NDL data is missing for certain isotopes.
This is indicated by the label stating "missing: ...". The letters stand for :
"C" means the "Capture" data is missing, "E" means the "elastic scattering" data
is missing and "I" means the "inelastic scattering" data is missing.

> ```{figure} ./_img/neutrons/neutron_jeff_vs_g4ndl.output.png

Reviewer 3: Consider scaling the cross-sections of the isotopes by their isotope fraction, such that it is clearer that the total corresponds to the sum of the individual contributions. 

Alternatively, if you want to highlight the cross sections of the individual isotopes separately, consider splitting this into four subplots, each comparing the JEFF and G4NDL data.

Author: I do like this idea. I did adjust it and it looks better i think

the simulation works, the total cross-section for neutrons in liquid Argon is
reconstructed from an idealized simulation. In this idealized simulation an orb
of 2 km radius is used. There are three worlds and three simulations, one for
each argon isotope. The density (and temperature) of the argon is that of liquid
argon. Every process is active and the Neutrons are sampled from a log-uniform
distribution from 0.01 eV to 10 MeV. To calculate the total cross-sections,
first the step length between two steps is derived, using the position of each
step. The mean of this value is taken as mean free path and used to derive the
total cross-section. The energy is given by the velocity of the particle at
pre-step. Due to the 2 km radius of the orb, geometric boundaries should not be
relevant. Additionally, a distance to surface check only counts those events,
that end up at least 10 m away from any boundary. Still, the results look like
there might be a bias from the simulation. This could be caused by non-physical
simulation processes limiting the step length.

:::{todo}

Investigate the effect of non-physical simulation processes on this

> cross-section reconstruction.

Reviewer 3: Idea on how to cross-check (not necessary for this PR):

Set up a thin target neutron beam experiment and count how many neutrons are unchanged in energy and direction on the other side. This way, you could extract the total cross-section by scanning through energy.

Author: I added the idea to the todo

:::

The plot shows the total weighted cross-section for natural argon according to
JEFF-3.3, G4NDL and using this reconstruction method.

```{figure} ./_img/neutrons/neutron_argon_summary.output.png
:width: 70%
```

Because some isotopes might be missing in G4NDL, it can be possible that the
simulation does not reproduce the G4NDL prediction. It might therefore be useful

> to investigate the cross-sections for the individual isotopes again:

Reviewer 3: Consider adding a sentence here that in G4NDL4.7.1 the cross-sections for Ar36 and Ar38 were removed and that their disagreement with JEFF is expected. Maybe also add a reference to the entry in the change logs: https://geant4.web.cern.ch/download/release-notes/notes-v11.2.2.txt

Author: I did not want to do this, as this validation suite should just run in the background for every remage and Geant4 version. So i did not want to add static information to the text, which will be outdated soon. Instead, i tried to make it visible in the plots, which are dynamicly produced. I mean thats the whole purpose of this validation suite.

If we just wanted to staticly check G4NDL4.7.1 that is a whole different thing

Reviewer 3: I guess it depends on who the validation suites should be written for and how they should be written. I understand that you would prefer it written so experts can quickly identify errors without having to read much boilerplate. If this were a purely technical document, I would agree with you.

However, I think it's better to consider a more "general" G4 programmer audience that doesn't know how hadronic/neutron physics is implemented in G4. Yes, some experts will review the tests to ensure everything is fine, but non-experts will do so as well, and they may raise unnecessary alarms. Additionally, a validation suite is the perfect place to educate people on these subjects, as understanding them is necessary to know what to expect. Therefore, we should not expect people to know how Neutron_HP works. They would also do not know where to find the cross-sections or why some of them might be missing. Spelling out these things would help them understand the expected disagreement. 

Finally, if not adjusted in the future, this kind of transient information about 4.7.1 would indeed be problematic, but since we will still have experts who will check these tests, it should be simple to remove the mention if the cross-sections are added again. It could be included in a warning section that would be difficult to overlook.

But again, this depends on one's opinion on how to write this documentation. If you are fine with it being a technical page only, then this is fine.

Reviewer 4: I agree @MoritzNeuberger but I think we should consider @EricMEsch point about the information becoming outdated. We definetly want the validation suite to have nice explanations but we should keep an eye on the future maintenance of this report. In principle you could check the version of G4NDL and add extra information if version == 4.7.1

Author: The issue with checking the version is that i can only do dynamic stuff in the python code, and the only current way to propagate this information is through the PNG that is created, which contains the plot.

AFAIK there is no other way currently, to relay dynamic information to the validation suite? As the text is static. Maybe you could add stuff to the `CMakeLists` but that would make it way more complicated again.

Author: I added a note about the isotopes missing now

```{subfigure} ABC
:subcaptions: above

:::{image} ./_img/neutrons/neutron_argon36_comparison.output.png
:width: 90%
:alt: argon-36 total cross-section.
:::

:::{image} ./_img/neutrons/neutron_argon38_comparison.output.png
:width: 90%
:alt: argon-38 total cross-section.
:::

:::{image} ./_img/neutrons/neutron_argon40_comparison.output.png
:width: 90%
:alt: argon-40 total cross-section.
:::

The total cross-sections of the three stable argon isotopes.
```

For cases where the G4NDL cross-section is missing, Geant4 might fall back to
use the cross-section of similar isotopes instead.

## Complex geometry

Last, we try to show some parameters simulating a more complex world, similar to
many particle physics experiments. Neutrons are sampled at 5 keV but every
process is active. Neutrons are simulated in a steel container filled with
liquid argon. The steel container itself is located within a water volume and
inside of the argon there is also a neutron moderator, which consists of

> hydrogen and is loaded with gadolinium. The goal of the geometry is to enable

Reviewer 3: I think it would be nice to have a small render of the geometry to better understand the setup and the relative expected contribution of the individual volumes' isotopes.

:width: 50%
```

While this pie-chart can not be compared to literature values, as it also
heavily depends on the neutron transport, it can be compared between _remage_
versions.

The neutron capture gamma de-excitation can be validated against literature
values. Hydrogen has a well-known singular gamma line at 2.22 MeV after a
neutron capture, making bugs very easy to spot. Gadolinium on the contrary, has
a very complex gamma line intensity, which can be compared to
[publications](https://link.springer.com/article/10.1140/epjc/s10052-023-11602-y).
The Q-Values of the neutron capture process is well known for all captures in
this setup. The expected Q-value according to the
[IAEA Nuclear Data Section](https://www-nds.iaea.org/) is shown in each plot for
the respective isotope. This can directly be compared with the orange line,
depicting the summed energy of all secondary particles resulting from the
neutron capture process in Geant4. In blue, the gamma line intensity is shown.

For every isotope there are always two plots shown. The plot on the left just
specifies the basic `G4Shielding` physicslist to the simulation without any

> extra modifications. The plot on the right instead additionally forces the

Reviewer 3: Rather than saying it is G4Shielding, specify that it is the Neutron_HP model within Shielding. Since it is also used by all other relevant hadronic physics lists, this makes clear that it is not a feature of just the Shielding physics lists, but of this specific model and therefore default physics list independent.

Reviewer 3: Maybe also add some references here on the specific gamma cascade models.
